app/training/[filename]/page.tsx

```diff
@@ -0,0 +1,70 @@
+import { getTestimonialsByCategories } from "@/helpers/getTestimonials";
+import { TODAY } from "@/hooks/useFetchEvents";
+import { useSEO } from "@/hooks/useSeo";
+import client from "@/tina/client";
+import { Metadata } from "next";
+import TrainingPage from ".";
+import { TinaClient } from "../../tina-client";
+
+export const dynamicParams = false;
+
+type GenerateMetaDataProps = {
+  params: { filename: string };
+  searchParams: { [key: string]: string | string[] | undefined };
+};
+
+export async function generateMetadata({
+  params,
+}: GenerateMetaDataProps): Promise<Metadata> {
+  const tinaProps = await getData(params.filename);
+  const seo = tinaProps.props.seo;
+
+  if (seo && !seo.canonical) {
+    seo.canonical = `${tinaProps.props.header.url}offices/${params.filename}`;
+  }
```

Comment:
I think you forgot to update this

```suggestion

  if (seo && !seo.canonical) {
    seo.canonical = `${tinaProps.props.header.url}training/${params.filename}`;
  }
```

Reply:
Done